Userland/Libraries/LibCrypto/PK/RSA.h

```diff
@@ -65,21 +96,41 @@ class RSAPrivateKey {
     Integer const& modulus() const { return m_modulus; }
     Integer const& private_exponent() const { return m_private_exponent; }
     Integer const& public_exponent() const { return m_public_exponent; }
+    Integer const& prime1() const { return m_p; }
+    Integer const& prime2() const { return m_q; }
+    Integer const& exponent1() const { return m_dp; }
+    Integer const& exponent2() const { return m_dq; }
+    Integer const& coefficient() const { return m_qinv; }
     size_t length() const { return m_length; }
-    void set_length(size_t length) { m_length = length; }
 
-    void set(Integer n, Integer d, Integer e)
+    ErrorOr<ByteBuffer> export_as_der() const
     {
-        m_modulus = move(n);
-        m_private_exponent = move(d);
-        m_public_exponent = move(e);
-        m_length = m_modulus.trimmed_length() * sizeof(u32);
+        ASN1::Encoder encoder;
+        TRY(encoder.write_constructed(ASN1::Class::Universal, ASN1::Kind::Sequence, [&]() -> ErrorOr<void> {
+            TRY(encoder.write(0x00u));              // version
+            TRY(encoder.write(m_modulus));          // n
+            TRY(encoder.write(m_public_exponent));  // e
+            TRY(encoder.write(m_private_exponent)); // d
+            TRY(encoder.write(m_p));                // p
+            TRY(encoder.write(m_q));                // q
+            TRY(encoder.write(m_dp));               // d mod (p-1)
+            TRY(encoder.write(m_dq));               // d mod (q-1)
+            TRY(encoder.write(m_qinv));             // q^-1 mod p
+            return {};
+        }));
+
+        return encoder.finish();
     }
 
 private:
     Integer m_modulus;
     Integer m_private_exponent;
     Integer m_public_exponent;
+    Integer m_p;
+    Integer m_q;
+    Integer m_dp;   // d mod (p-1)
+    Integer m_dq;   // d mod (q-1)
+    Integer m_qinv; // q^-1 mod p
```

Comment:
should we save these as the names they have in the interface/the RSAPrivateKey ASN.1 thingy, and keep the RSA-spec-names as comments?

Reply:
I don't really mind either, I'll rename to the spec names.